prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -4,10 +4,26 @@
 
 import redfish
 import redfish_utilities
-from redfish.rest.v1 import HttpClient, InvalidCredentialsError, SessionCreationError
+from cachetools import TTLCache, cached
+from redfish.rest.v1 import (
+    HttpClient,
+    InvalidCredentialsError,
+    RetriesExhaustedError,
+    SessionCreationError,
+)
 
 logger = getLogger(__name__)
 
+# timeout in seconds for initial connection
+REDFISH_CLIENT_TIMEOUT = 3
```

Comment:
**nitpick:** 

Instead of putting settings here, how about loading them from the config?
We can provide a default value also.

Reply:
I think that it's best we keep these configs directly in `redfish.py` since it would be easier to refer them and change it here rather than going to a different file. 

The configs for `sasircu.py` collector are also inside the same file.

Reply:
I would like to change the settings with config instead of the source code change. It will be useful when debugging or turning.